File: SolutionSourceCode/30796번_실버_4_gahui and sousenkyo 4.py
import sys
import math

def actual_code():
    v, k = map(int, sys.stdin.readline().split())

    votes = []
    current_vote = v
    while current_vote > 0:
        # Add k values
        for i in range(k):
            if current_vote > 0:
                votes.append(current_vote)
                current_vote -= 1

        # Skip next k values
        current_vote -= k

    sys.stdout.write(str(len(votes)) + '\n')
    for vote in votes:
        sys.stdout.write(str(vote) + '\n')

if __name__ == '__main__':
    actual_code()

File: SolutionSourceCode/30796번_실버_4_gahui and sousenkyo 4.cs
using System;
using System.Collections.Generic;
using System.IO;

class Program
{
    static void Main()
    {
        StreamReader reader = new StreamReader(new BufferedStream(Console.OpenStandardInput()));
        StreamWriter writer = new StreamWriter(new BufferedStream(Console.OpenStandardOutput()));
        string[] input = reader.ReadLine().Split(' ');
        long v = long.Parse(input[0]);
        long k = long.Parse(input[1]);

        List<long> votes = new List<long>();
        long currentVote = v;

        while (currentVote > 0)
        {
            // Add k values
            for (long i = 0; i < k; i++)
            {
                if (currentVote > 0)
                {
                    votes.Add(currentVote);
                    currentVote -= 1;
                }
            }

            // Skip next k values
            currentVote -= k;
        }

        writer.WriteLine(votes.Count.ToString());
        foreach (long vote in votes)
        {
            writer.WriteLine(vote.ToString());
        }

        reader.Close();
        writer.Close();
    }
}